### ISSUE TITLE
Case-insensitive fuzzy completions?

### ISSUE THREAD

**Issue by seanlavelle** [fish-shell/fish-shell#1196]
I just updated to 2.1.0 and I love the fuzzy completions ( https://github.com/fish-shell/fish-shell/issues/568 )
but they are case-sensitive when I would prefer them to be case-insensitive.

For example:

```
mkdir fooBar
cd fb<tab> (nothing happens, sad face)
cd fB<tab> (it completes fooBar)
```

**Comment by mqudsi** [comment]
dfe6bc5 improves fuzzy completions by supporting case-insensitive substring searches as a separate (lower-priority) after case-sensitive substring searches.

**Comment by MatisseHack** [comment]
> It'd be better to weight a list of results and cycle through them instead of just take the "closest" completion and only give you that.

I just ran into this behavior as well with a completion that returns multiple matches with varied casing. e.g.

```shell
complete -c foo -a "(echo -e bar\nBAZ)"
```

I expected both `bar` and `BAZ` to be suggested when typing `foo b`, but `bar` gets selected automatically without any option to choose `BAZ`.

**Comment by krobelus** [comment]
> I expected both `bar` and `BAZ` to be suggested when typing `foo b`, but `bar` gets selected automatically without any option to choose `BAZ`.

@MatisseHack what happens here is that fish prioritizes completions that can be added at the end of the token over completions that need to replace the token. (If you type `foo a` you'll get both.)
So this is a different issue, https://github.com/fish-shell/fish-shell/issues/7944
The current behavior might be right, I'm not sure.

**Comment by MatisseHack** [comment]
Good call! Thanks for pointing me in the right direction, I've up-voted that issue.
